From 83d368bcd58ef768e349d5926065eb91156d9cc9 Mon Sep 17 00:00:00 2001
From: Prem Sichanugrist <s@sikachu.com>
Date: Wed, 23 Feb 2011 02:25:38 +0700
Subject: [PATCH 1/2] Make sure that we set `env["action_dispatch.show_exceptions"]`

This has been used by `show_exception` middleware even the setting wasn't get passed to `env` hash.
---
 railties/lib/rails/application.rb               |    3 ++-
 railties/test/application/configuration_test.rb |   15 +++++++++++++++
 2 files changed, 17 insertions(+), 1 deletions(-)

diff --git a/railties/lib/rails/application.rb b/railties/lib/rails/application.rb
index 9cb3a0f..fb0f6fe 100644
--- a/railties/lib/rails/application.rb
+++ b/railties/lib/rails/application.rb
@@ -122,7 +122,8 @@ module Rails
       @env_config ||= super.merge({
         "action_dispatch.parameter_filter" => config.filter_parameters,
         "action_dispatch.secret_token" => config.secret_token,
-        "action_dispatch.asset_path" => nil
+        "action_dispatch.asset_path" => nil,
+        "action_dispatch.show_exceptions" => config.action_dispatch.show_exceptions
       })
     end
 
diff --git a/railties/test/application/configuration_test.rb b/railties/test/application/configuration_test.rb
index 4f4b7be..044fd2a 100644
--- a/railties/test/application/configuration_test.rb
+++ b/railties/test/application/configuration_test.rb
@@ -318,5 +318,20 @@ module ApplicationTests
 
       assert ActionView::Resolver.caching?
     end
+
+    test "config.action_dispatch.show_exceptions is sent in env" do
+      make_basic_app do |app|
+        app.config.action_dispatch.show_exceptions = true
+      end
+
+      class ::OmgController < ActionController::Base
+        def index
+          render :text => env["action_dispatch.show_exceptions"]
+        end
+      end
+
+      get "/"
+      assert_equal 'true', last_response.body
+    end
   end
 end
-- 
1.7.4.1

